### ISSUE TITLE
Consider removing Storage SDK version cap

### ISSUE THREAD

**Issue by mathewc** [Azure/azure-webjobs-sdk#631]
Now that the 6.1 incompatibility has been addressed in 6.2. http://blog.davidebbo.com/2011/01/nuget-versioning-part-2-core-algorithm.html. See the "When an upper bound makes sense" section.

**Comment by Itertek** [comment]
Yes please - I've got way too many other dependencies to revert to storage 5.0.2

**Comment by ctolkien** [comment]
:+1:

**Comment by markjbrown** [comment]
PLEASE PLEASE PLEASE update this to 6.2.0. I am building a sample app that combines Azure Storage Data Movement Library 0.2 which requires Azure Storage 6.2.0. 

Right now I have a solution with multiple versions of WebJobs and Azure Storage and all their dependencies. It's a mess :)

Thanks!

**Comment by davidebbo** [comment]
The idea is to simply remove the NuGet cap, which will allow it to work with any higher version. @mathewc maybe we can make a 1.1.2 service release that just changes this?

**Comment by tedvanderveen** [comment]
+1

**Comment by Itertek** [comment]
and another +1 for me, it's holding back a project I'm working on and very quickly I'll have to make a choice between writing something non-optimal myself or waiting for this release.

**Comment by mathewc** [comment]
I've removed the version cap in this commit: https://github.com/Azure/azure-webjobs-sdk/commit/50f182b019838cffd0ae1005e7d9a380740c4189. I've pushed these updated versions to our myget feed (1.1.2-alpha-10252).

**Comment by markjbrown** [comment]
Do you have an ETA for when that version would get pushed to GA?

**Comment by Itertek** [comment]
Thanks Mathew,  excellent news...

**Comment by mathewc** [comment]
Need to discuss GA timing with the team, I'll report back here when I know more.

**Comment by davidebbo** [comment]
Please see [this page](https://github.com/Azure/azure-webjobs-sdk/wiki/%22Nightly%22-Builds) for info on using the MyGet daily feed..

**Comment by jurjen74** [comment]
This has been for a few months now. 1.1.1 is still the last one in nuget and it still has the cap.
Another packages needs the newest 7.0.0 so we have a problem now... Please tell me that a newer version will be available soon....

**Comment by hsteele-sonovate** [comment]
Would be great to get some feedback on the due date for the availability of the uncapped package on the regular feed, if possible!

**Comment by chriswill** [comment]
I went looking for a pre-release Nuget package from Microsoft.Azure.WebJobs to address this issue but couldn't find 1.1.2-alpha-10252.  

Does it still exist?  Would also love to see the cap removed.  It's a blocking issue for me.

**Comment by mathewc** [comment]
Update: we've just released version 1.1.2 of the SDK on Nuget.org. This version is a hotfix to 1.1.1 that removes the Storage version cap that we had in place. So if you upgrade to 1.1.2 you should now be unblocked to upgrade to later storage versions. You should be fine upgrading to 6.2 - I did a test pass on that and it looked good. I haven't tried 7.0 yet.

**Comment by chriswill** [comment]
Thank you so much!  Already applied and getting set to deploy.  I'm running 7.0, so I'll post an issue if I encounter any problems.

**Comment by markjbrown** [comment]
Same here. Just updated my project and also now using Azure Storage 7.0.

Thanks so much for unblocking this.

**Comment by MatthewSteeples** [comment]
We're having an issue with 7:

```
Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Exception while executing function: Program.GenerateExcelFile ---> System.InvalidOperationException: Exception binding parameter 'tblMessage' ---> System.MissingMethodException: Method not found: 'Microsoft.WindowsAzure.Storage.Table.TableOperation Microsoft.WindowsAzure.Storage.Table.TableOperation.Retrieve(System.String, System.String)'.
   at Microsoft.Azure.WebJobs.Host.Storage.Table.StorageTableOperation.Retrieve[TElement](String partitionKey, String rowKey)
   at Microsoft.Azure.WebJobs.Host.Tables.TableEntityArgumentBinding`1.<BindAsync>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Triggers.TriggeredFunctionBinding`1.<BindCoreAsync>d__7.MoveNext()
   --- End of inner exception stack trace ---
   at Microsoft.Azure.WebJobs.Host.Executors.DelayedException.Throw()
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithWatchersAsync>d__31.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__2c.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__13.MoveNext()
   --- End of inner exception stack trace ---
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLoggingAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<TryExecuteAsync>d__1.MoveNext()
```

Looks like the method that was removed in 6 then re-added in 6.2 has been removed again.

**Comment by mathewc** [comment]
Commit #709 has moved the SDK to the latest packages of all our main dependencies. On our myget feed as 2.0.0-alpha-10313. We're going to try to get a prerelease up on Nuget soon.

**Comment by mathewc** [comment]
FYI: We've released the [2.0.0-beta1](https://www.nuget.org/packages/Microsoft.Azure.WebJobs/2.0.0-beta1) prerelease on Nuget. In this version, we've moved completely to the latest Storage SDK version.
